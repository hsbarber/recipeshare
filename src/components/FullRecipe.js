import React from 'react';
import PropTypes from 'prop-types';
import AuthUserContext from '../components/AuthUserContext';
import Edit from './Edit';

function FullRecipe(props){

      const recipes = props.recipes;
      const title = props.match.params.title;

      const rec = recipes.filter(rec => {
        if(rec.title.split(' ').join('') === title) {
            return rec;
        }});

        const List = rec.map((recipe) => {
          return (
            props.canEdit & props.editable === recipe.id ?
              <Edit {...props} />
              :
                      <div key={recipe.id}>
                      <AuthUserContext.Consumer>
                        {(authUser) => authUser.email === recipe.email && <button className="fullRecipe--edit" onClick={() => props.getRecipe(recipe.id)}>Edit</button>}
                      </AuthUserContext.Consumer>
                        {/* <Link to={routes.EDIT}><button className="fullRecipe--edit" onClick={() => props.getRecipe(recipe.id)}>Edit</button></Link> */}
                        <div  className="fullRecipe">


                              <div className="fullRecipe--tBlock">
                                <div className="fullRecipe--tBlock-box">
                                  <div className="titleBox">
                                    <h3>{recipe.title}</h3>
                                    <p><span>Category</span><span>{recipe.category}</span></p>
                                    {recipe.recipeTime && <p><span>Time to Cook</span><span>{recipe.recipeTime}</span></p>}
                                    {recipe.link && <a href={recipe.link}>Link to Original Recipe</a>}
                                    <p>{recipe.notes}</p>
                                  </div>
                                </div>
                                {recipe.imageURL ?
                                  <div className="fullRecipe--tBlock-image" style= { {
                                    backgroundImage:
                                    `url(${recipe.imageURL })` } }>
                                  </div>
                                  :
                                  <div className="fullRecipe--tBlock-image" style= { {
                                    backgroundColor:
                                    `#627bc0` } }>
                                      <h3>No Image Available</h3>
                                  </div>
                              }
                              </div>


                              <div  className="fullRecipe--bBlock">
                                  <div className="fullRecipe--bBlock-ingredients">
                                    <h3>Ingredients</h3>
                                    {recipe.ingredients.map(ingredient =>
                                      <p key = {ingredient.id}>{ingredient.content}</p>
                                    )}
                                  </div>
                                  <div className="fullRecipe--bBlock-steps">
                                    <h3>Steps</h3>
                                    {recipe.steps.map((step, index) =>
                                      <p key = {step.id}><span>{index + 1}.</span> {step.content}</p>
                                    )}
                                  </div>
                              </div>
                        </div>
                      </div>

        )
    })
    return (

      <React.Fragment>

          {List}

      </React.Fragment>
    )

}
FullRecipe.propTypes = {
  recipes: PropTypes.array,
  title: PropTypes.object,
  recipe: PropTypes.arrayOf(PropTypes.string)
}
export default FullRecipe;

// const name = rec.map(r => r.title);
//       const Ingredients = rec.map(r => r.ingred.map(food =>
//           <p key = {food.id}>{food.text}</p>
//         )
//       )
//       const steps = rec.map(r => r.steps);
//       console.log(name);




 // const recipes = this.props.recipes;
 //      const title = this.props.match.params.title;
 //      function Search (id) {
 //        const isRecipe = r => r.title === id
 //        return recipes.find(isRecipe)
 //      }
 //      const rec = recipes.Search(title);



// function get (recipe) {
//     const isRecipe = r => r.title === recipe
//     return {props.recipes.find(isRecipe)}
//   }
//     const recipe = get(props.match.params.title);
//         if (!recipe) {
//           return <div>Sorry, but the recipe was not found</div>
//         }