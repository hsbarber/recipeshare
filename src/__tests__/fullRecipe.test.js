import * as React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import '../enzyme';
import FullRecipe from '../components/FullRecipe';
import AuthUserContext from '../components/AuthUserContext';

const fakeItem = [
  {
    category: 'Mediterranean',
    email: 'haganb3@gmail.com',
    id: '-LKsBjUcn-7KrbxFIlJa',
    imageURL:
      'https://firebasestorage.googleapis.com/v0/b/recipes-cc787.appspot.com/o/images%2F3a92e118-0760-4bfc-92b6-d1176a1889d4.jpg?alt=media&token=f9e3c5e5-6766-41be-a35e-d026f1075826',
    ingredients: [
      { content: 'butter', id: 1535319837489 },
      { content: 'bread', id: 1535319837488 },
      { content: 'cheese', id: 1535319837487 },
    ],
    link: undefined,
    notes: 'easy soup with squash and butter and sage.',
    recipeTime: undefined,
    steps: [
      { content: 'heat up', id: 1535319837491 },
      { content: 'STIR', id: 1535319837492 },
      { content: 'serve', id: 1535319837493 },
    ],
    time: 1535319939642,
    title: 'Roasted Butternut Squash Soup',
    user: 'hagan',
  },
];
const getDefaultProps = () => ({
  match: {
    isExact: true,
    params: {
      title: 'RoastedButternutSquashSoup',
      path: '/:title',
      url: '/RoastedButternutSquashSoup',
    },
  },
  user: 'hagan',
  authUser: { email: 'haganb3@gmail.com' },
});

describe('RecipeItem', () => {
  const { match, user, authUser } = getDefaultProps();
  it('shallow renders without crashing', () => {
    shallow(<FullRecipe recipes={fakeItem} match={match} />);
  });
  it('recipe matches snapshot', () => {
    const wrapper = shallow(
      <FullRecipe recipes={fakeItem} match={match} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Edit button available when user logged in and user equals recipe user', () => {
    const wrapper = mount(
      <MemoryRouter>
        <AuthUserContext.Provider value={{ authUser }}>
          <FullRecipe recipes={fakeItem} match={match} user={user} />
        </AuthUserContext.Provider>
      </MemoryRouter>
    );

    expect(wrapper.find('.fullRecipe--edit').exists()).toEqual(true);
    wrapper.unmount();
  });
  it('When user is different, edit button not visible', () => {
    const diffUser = 'sam';
    const wrapper = mount(
      <MemoryRouter>
        <AuthUserContext.Provider value={{ authUser }}>
          <FullRecipe recipes={fakeItem} match={match} user={diffUser} />
        </AuthUserContext.Provider>
      </MemoryRouter>
    );

    expect(wrapper.find('.fullRecipe--edit').exists()).toEqual(false);
    wrapper.unmount();
  });
});
