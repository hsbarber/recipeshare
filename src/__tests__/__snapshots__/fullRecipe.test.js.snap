// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeItem recipe matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FullRecipe
    match={
      Object {
        "isExact": true,
        "params": Object {
          "path": "/:title",
          "title": "RoastedButternutSquashSoup",
          "url": "/RoastedButternutSquashSoup",
        },
      }
    }
    recipes={
      Array [
        Object {
          "category": "Mediterranean",
          "email": "haganb3@gmail.com",
          "id": "-LKsBjUcn-7KrbxFIlJa",
          "imageURL": "https://firebasestorage.googleapis.com/v0/b/recipes-cc787.appspot.com/o/images%2F3a92e118-0760-4bfc-92b6-d1176a1889d4.jpg?alt=media&token=f9e3c5e5-6766-41be-a35e-d026f1075826",
          "ingredients": Array [
            Object {
              "content": "butter",
              "id": 1535319837489,
            },
            Object {
              "content": "bread",
              "id": 1535319837488,
            },
            Object {
              "content": "cheese",
              "id": 1535319837487,
            },
          ],
          "link": undefined,
          "notes": "easy soup with squash and butter and sage.",
          "recipeTime": undefined,
          "steps": Array [
            Object {
              "content": "heat up",
              "id": 1535319837491,
            },
            Object {
              "content": "STIR",
              "id": 1535319837492,
            },
            Object {
              "content": "serve",
              "id": 1535319837493,
            },
          ],
          "time": 1535319939642,
          "title": "Roasted Butternut Squash Soup",
          "user": "hagan",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
      "enter": Object {
        "opacity": 1,
        "transform": "translate3d(0,0px,0)",
      },
      "from": Object {
        "opacity": 0,
        "transform": "translate3d(0,60px,0)",
      },
      "items": Array [
        <div>
          <div
            className="fullRecipe"
          >
            <div
              className="fullRecipe--tBlock"
            >
              <div
                className="fullRecipe--tBlock-box"
              >
                <div
                  className="titleBox"
                >
                  <h3>
                    Roasted Butternut Squash Soup
                  </h3>
                  <p>
                    <span>
                      Category
                    </span>
                    <span>
                      Mediterranean
                    </span>
                  </p>
                  <p>
                    easy soup with squash and butter and sage.
                  </p>
                  <Context.Consumer>
                    [Function]
                  </Context.Consumer>
                </div>
              </div>
              <div
                className="fullRecipe--tBlock-image"
                style={
                  Object {
                    "backgroundImage": "url(https://firebasestorage.googleapis.com/v0/b/recipes-cc787.appspot.com/o/images%2F3a92e118-0760-4bfc-92b6-d1176a1889d4.jpg?alt=media&token=f9e3c5e5-6766-41be-a35e-d026f1075826)",
                  }
                }
              />
            </div>
            <div
              className="fullRecipe--bBlock"
            >
              <div
                className="fullRecipe--bBlock-ingredients"
              >
                <h3>
                  Ingredients
                </h3>
                <p>
                  butter
                </p>
                <p>
                  bread
                </p>
                <p>
                  cheese
                </p>
              </div>
              <div
                className="fullRecipe--bBlock-steps"
              >
                <h3>
                  Steps
                </h3>
                <p>
                  <span>
                    1
                    .
                  </span>
                   
                  heat up
                </p>
                <p>
                  <span>
                    2
                    .
                  </span>
                   
                  STIR
                </p>
                <p>
                  <span>
                    3
                    .
                  </span>
                   
                  serve
                </p>
              </div>
            </div>
          </div>
        </div>,
      ],
      "keys": [Function],
      "leave": Object {
        "opacity": 0,
        "transform": "translate3d(0,-60px,0)",
      },
      "reset": false,
      "unique": false,
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
        "enter": Object {
          "opacity": 1,
          "transform": "translate3d(0,0px,0)",
        },
        "from": Object {
          "opacity": 0,
          "transform": "translate3d(0,60px,0)",
        },
        "items": Array [
          <div>
            <div
              className="fullRecipe"
            >
              <div
                className="fullRecipe--tBlock"
              >
                <div
                  className="fullRecipe--tBlock-box"
                >
                  <div
                    className="titleBox"
                  >
                    <h3>
                      Roasted Butternut Squash Soup
                    </h3>
                    <p>
                      <span>
                        Category
                      </span>
                      <span>
                        Mediterranean
                      </span>
                    </p>
                    <p>
                      easy soup with squash and butter and sage.
                    </p>
                    <Context.Consumer>
                      [Function]
                    </Context.Consumer>
                  </div>
                </div>
                <div
                  className="fullRecipe--tBlock-image"
                  style={
                    Object {
                      "backgroundImage": "url(https://firebasestorage.googleapis.com/v0/b/recipes-cc787.appspot.com/o/images%2F3a92e118-0760-4bfc-92b6-d1176a1889d4.jpg?alt=media&token=f9e3c5e5-6766-41be-a35e-d026f1075826)",
                    }
                  }
                />
              </div>
              <div
                className="fullRecipe--bBlock"
              >
                <div
                  className="fullRecipe--bBlock-ingredients"
                >
                  <h3>
                    Ingredients
                  </h3>
                  <p>
                    butter
                  </p>
                  <p>
                    bread
                  </p>
                  <p>
                    cheese
                  </p>
                </div>
                <div
                  className="fullRecipe--bBlock-steps"
                >
                  <h3>
                    Steps
                  </h3>
                  <p>
                    <span>
                      1
                      .
                    </span>
                     
                    heat up
                  </p>
                  <p>
                    <span>
                      2
                      .
                    </span>
                     
                    STIR
                  </p>
                  <p>
                    <span>
                      3
                      .
                    </span>
                     
                    serve
                  </p>
                </div>
              </div>
            </div>
          </div>,
        ],
        "keys": [Function],
        "leave": Object {
          "opacity": 0,
          "transform": "translate3d(0,-60px,0)",
        },
        "reset": false,
        "unique": false,
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
